cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -2520,122 +2604,139 @@ Slice::Python::CodeVisitor::writeDocstring(const OperationPtr& op, DocstringMode
     {
         case DocSync:
         case DocAsync:
-        case DocAsyncBegin:
         case DocDispatch:
             needArgs = true;
             break;
-        case DocAsyncEnd:
         case DocAsyncDispatch:
             needArgs = true;
             break;
     }
 
     if (needArgs)
     {
-        _out << nl << "Arguments:";
-        for (vector<string>::iterator q = inParams.begin(); q != inParams.end(); ++q)
+        if (!comment.description.empty())
+        {
+            _out << nl;
+        }
+
+        _out << nl << "Parameters";
+        _out << nl << "----------";
+        for (const auto& param : inParams)
         {
-            string fixed = fixIdent(*q);
-            _out << nl << fixed << " -- ";
-            StringMap::const_iterator r = comment.params.find(*q);
+            string fixed = fixIdent(param->name());
+            _out << nl << fixed << " : " << typeToDocstring(param->type(), param->optional());
+            StringMap::const_iterator r = comment.params.find(param->name());
             if (r == comment.params.end())
             {
                 r = comment.params.find(fixed); // Just in case.
             }
             if (r != comment.params.end())
             {
-                _out << r->second;
+                _out << nl << "    " << r->second;
             }
         }
-        if (mode == DocAsyncBegin)
-        {
-            _out << nl << "_response -- The asynchronous response callback." << nl
-                 << "_ex -- The asynchronous exception callback." << nl << "_sent -- The asynchronous sent callback.";
-        }
-        if (mode == DocSync || mode == DocAsync || mode == DocAsyncBegin)
+        if (mode == DocSync || mode == DocAsync)
         {
             const string contextParamName = getEscapedParamName(op, "context");
-            _out << nl << contextParamName << " -- The request context for the invocation.";
+            _out << nl << contextParamName << " : Ice.Context";
+            _out << nl << "    The request context for the invocation.";
         }
         if (mode == DocDispatch || mode == DocAsyncDispatch)
         {
             const string currentParamName = getEscapedParamName(op, "current");
-            _out << nl << currentParamName << " -- The Current object for the invocation.";
+            _out << nl << currentParamName << " : Ice.Current";
+            _out << nl << "    The Current object for the invocation.";
```

Comment:
We should not mix-up invocation and dispatch.
client-side including client-side context => invocation
server-side including server-side current => dispatch

Reply:
Fixed